flows/runs/run.go

```diff
@@ -117,12 +116,15 @@ func (r *flowRun) Input() flows.Input         { return r.input }
 func (r *flowRun) SetInput(input flows.Input) { r.input = input }
 
 func (r *flowRun) ApplyEvent(s flows.Step, action flows.Action, event flows.Event) error {
+	if s != nil {
+		event.SetStepUUID(s.UUID())
+	}
```

Comment:
so that events know what step they're being executed on and can set node_uuid correctly on results that they create